patchwork-extensions-blockentity/src/main/java/net/minecraftforge/common/extensions/IForgeTileEntity.java

```diff
@@ -0,0 +1,197 @@
+import net.minecraft.block.SkullBlock;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.ClientConnection;
+import net.minecraft.network.packet.s2c.play.BlockEntityUpdateS2CPacket;
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.util.math.Box;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+public interface IForgeTileEntity extends ICapabilitySerializable<CompoundTag> {
+	/**
+	 * Sometimes default render bounding box: infinite in scope. Used to control rendering on {@link TileEntitySpecialRenderer}.
+	 */
+	Box INFINITE_EXTENT_AABB = new net.minecraft.util.math.Box(Double.NEGATIVE_INFINITY, Double.NEGATIVE_INFINITY, Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY);
+
+	default BlockEntity getTileEntity() {
+		return (BlockEntity) this;
+	}
+
+	@Override
+	default void deserializeNBT(CompoundTag nbt) {
+		getTileEntity().fromTag(nbt);
+	}
+
+	@Override
+	default CompoundTag serializeNBT() {
+		CompoundTag ret = new CompoundTag();
+		getTileEntity().toTag(ret);
+		return ret;
+	}
+
+	/**
+	 * Called when you receive a TileEntityData packet for the location this
+	 * TileEntity is currently in. On the client, the NetworkManager will always
+	 * be the remote server. On the server, it will be whomever is responsible for
+	 * sending the packet.
+	 *
+	 * @param net The NetworkManager the packet originated from
+	 * @param pkt The data packet
+	 */
+	default void onDataPacket(ClientConnection net, BlockEntityUpdateS2CPacket pkt) {
+	}
+
+	/**
+	 * Called when the chunk's TE update tag, gotten from {@link #getUpdateTag()}, is received on the client.
+	 *
+	 * <p>Used to handle this tag in a special way. By default this simply calls {@link #readFromNBT(NBTTagCompound)}.
+	 *
+	 * @param tag The {@link NBTTagCompound} sent from {@link #getUpdateTag()}
+	 */
+	default void handleUpdateTag(CompoundTag tag) {
+		getTileEntity().fromTag(tag);
+	}
+
+	/**
+	 * Gets a {@link NBTTagCompound} that can be used to store custom data for this tile entity.
+	 * It will be written, and read from disc, so it persists over world saves.
+	 *
+	 * @return A compound tag for custom data
+	 */
+	CompoundTag getTileData();
+
+	default void onChunkUnloaded() {
+	}
+
+	/**
+	 * Called when this is first added to the world (by {@link World#addTileEntity(TileEntity)}).
+	 * Override instead of adding {@code if (firstTick)} stuff in update.
+	 */
+	default void onLoad() {
+		requestModelDataUpdate();
+	}
+
+	/**
+	 * Return an {@link AxisAlignedBB} that controls the visible scope of a {@link TileEntitySpecialRenderer} associated with this {@link TileEntity}
+	 * Defaults to the collision bounding box {@link Block#getCollisionBoundingBoxFromPool(World, int, int, int)} associated with the block
+	 * at this location.
+	 *
+	 * @return an appropriately size {@link AxisAlignedBB} for the {@link TileEntity}
+	 */
+	// @OnlyIn(Dist.CLIENT)
```

Comment:
why comment this out?

Reply:
Fabric's @Environment?

Reply:
does the patcher transform @OnlyIn to @Environment?

Reply:
yes